lib/onelogin/ruby-saml/response.rb

```diff
@@ -51,6 +53,21 @@ def initialize(response, options = {})
 
         @response = decode_raw_saml(response)
         @document = XMLSecurity::SignedDocument.new(@response, @errors)
+
+        return unless assertion_encrypted?
```

Comment:
why not clone the object inside `decrypt_assertion_from_document` and pass `document` or `response` to that same method?